EXception :

There are two types of Exception :

1)Checked Exception: The exceptions which are checked by compiler is called as Checked exception.
Real Time Example - HallTicketMissedException

Example :

Example :

package checkedException;

import java.io.FileNotFoundException;
import java.io.FileReader;

public class Test {

	public static void main(String[] args) {
		
		try {
			FileReader fr=new FileReader("abc.txt");
		} catch (FileNotFoundException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		}
	}
	
}



2)Unchecked Exception:The exceptions which are not checked by compiler is called as UnChecked exception.
Real Time Example: BombBlastException.

Note :All Exceptions will occur at run time.

Example :


package uncheckedException;

public class Demo {

	public static void main(String[] args) {
		
		int a=10;
		int b=0;
		try {
		int c=a/b;
		}
		catch(ArithmeticException e) {
			
		}
	}
}



By using following keyword we can handle Exceptions.

1)try
2)catch
3)throw
4)throws
5)finally

1)try catch Block:

Inside try block there will be a risky code.

syntax:

try{

risky code
}
catch(){

}

1)Once Exception occurs in try block then remaining line of code will not be executed.

Example :


package typesOfException;

public class ArithmeticExceptionDemo {

	public static void main(String[] args) {
		System.out.println("Main method Started ");

		int a = 10;
		int b = 0;
		try {
			System.out.println("try block started");
			int c = a / b;
            System.out.println("Try block ended");
		} catch (ArithmeticException e) {
          System.out.println("Catch Block");
		}
		System.out.println("Welcome to bikkad It");
		System.out.println("I want to beacome java Developer");

	}

}



2) If there is no exception in try block then catch block will be not executed.

package typesOfException;

public class ArithmeticExceptionDemo {

	public static void main(String[] args) {
		System.out.println("Main method Started ");

		int a = 10;
		int b = 2;
		try {
			System.out.println("try block started");
			int c = a / b;
            System.out.println("Try block ended");
		} catch (ArithmeticException e) {
          System.out.println("Catch Block");
		}
		System.out.println("Welcome to bikkad It");
		System.out.println("I want to beacome java Developer");

	}

}



































